code/game/gamemodes/traitor/double_agents.dm

```diff
@@ -18,33 +18,134 @@
 	var/list/target_list = list()
 	var/list/late_joining_list = list()
 
+
 /datum/game_mode/traitor/internal_affairs/post_setup()
 	var/i = 0
 	for(var/datum/mind/traitor in traitors)
 		i++
 		if(i + 1 > traitors.len)
 			i = 0
-		target_list[traitor] = traitors[i + 1]
+		target_list[traitor] = traitors[i+1]	
 	..()
 
+/datum/objective/assassinate/internal/proc/give_pinpointer()
+	if(owner && owner.current)
+		if(ishuman(owner.current))
+			var/mob/living/carbon/human/H = owner.current
+			var/list/slots = list ("backpack" = slot_in_backpack)
+			var/obj/item/weapon/pinpointer/internal/pinpointer = new
+			pinpointer.owner=owner
+			H.equip_in_one_of_slots(pinpointer, slots)
+
+/proc/is_internal_objective(datum/objective/O)
+	return (istype(O, /datum/objective/assassinate/internal)||istype(O, /datum/objective/destroy/internal))
+
+
+/proc/steal_targets(datum/mind/owner,datum/mind/victim)
+	if(!owner.current||owner.current.stat==DEAD) //Should already be guaranteed if this is only called from steal_targets_timer_func, but better to be safe code than sorry code 
+		return
+	var/traitored = FALSE
+	var/failed_traitored = FALSE
+	for(var/objective_ in victim.objectives)
+		if(istype(objective_, /datum/objective/assassinate/internal))
+			var/datum/objective/assassinate/internal/objective = objective_
+			if(objective.target==owner)
+				to_chat(owner.current,"<B><font size=3 color=red> Now that all the loyalist agents have been purged, your syndicate sleeper training activates - YOU ARE THE TRAITOR! You now have no limits on collateral damage.</font></B>")

```

Comment:
I agree, swapping your motivations like this is weird and kor's suggestion feels better.